### PR TITLE
Use concat for scipp-0.9

### PR DESCRIPTION
**Pull request by SimonHeybrock**
This should also significantly (or massively, if many banks) speed up `load_nexus` for Nexus files with multiple banks.

### CODE REVIEW

File: tests/test_load_nexus.py [hunk 1]
```diff
@@ -1815,7 +1815,9 @@ def test_load_raw_detector_data_from_nexus_file(load_function: Callable):
         end=sc.array(dims=["pulse"], values=[3, 3, 5], dtype=sc.dtype.int64),
     )
 
-    expected = sc.DataArray(data=binned,
+    # Even if there is just a single NXevent_data entry in the file the
+    # output has a `bank` dimension, for consistency.
+    expected = sc.DataArray(data=sc.concat([binned], 'bank'),
```
Comment (jl-wynen) [inline tests/test_load_nexus.py:1820]: Not sure if this helps with usability? It means that people with a single bank always have to slice out that dimension.

Comment (SimonHeybrock) [reply tests/test_load_nexus.py:1820]: I think there are two cases:
1. Load a specific NXevent_data (not supported yet). Then indeed there should not be a `bank` dimension.
2. Load entire file. Typically files have an unknown number of banks, and even the same instrument may have a fluctuating number of banks, e.g., when one is broken and removed, or new ones added.

For the latter case, writing scripts that don't break is easier if we consistently include the `bank` dimension here?

**Comment by jl-wynen** [comment]
There might be overlap with  #208

**Comment by SimonHeybrock** [comment]
> There might be overlap with #208

There shouldn't be, I checked before.

**Review by nvaytet — APPROVED** [review]

**Pull request merged**